### ISSUE TITLE
Update to Boot 2.0.0.M1

### ISSUE THREAD

**Issue by rwinch** [spring-projects/spring-security#4339]
### Summary

Update to Boot 2.0.0.M1

**Comment by tha2015** [comment]
Hi Rob,

First of all, thank you for the great work (especially OpenID Connect 1.0 feature).

It looks like the Spring Boot sample is still using Spring Boot 1.5.3.RELEASE (see spring-security/samples/boot/gradle/dependency-management.gradle). 

I'm trying to implement an OpenID Connect based app using Spring security 5.0.0 M1 on Spring Boot 2.0.0.M1 and I've got errors when compiling ClientRegistrationAutoConfiguration.java (from the Spring boot sample)

Specifically there are 2 classes  which are being used by org.springframework.boot.autoconfigure.security.oauth2.client.ClientRegistrationAutoConfiguration were removed from Spring Boot 2.0.0.M1:

  org.springframework.boot.bind.PropertySourcesBinder 
  org.springframework.boot.bind.RelaxedPropertyResolver

Thanks,
-Thai

**Comment by rwinch** [comment]
Thanks for reaching out. 

This is a question for the Boot team since they manage the auto configuration and the missing classes.

**Comment by tha2015** [comment]
Thanks Rob. But that Boot sample belong to Spring Security repository, is it correct?
(Adding Joe Grandja who worked on the sample to the discussion)
@jgrandja : Hi Joe.  Do you know when will Spring Security samples be updated to use Spring Boot 2.0 builds? Thanks.

**Comment by rwinch** [comment]
@tha2015 I'm sorry I was confused which auto configuration. We plan to update by M2 which is [due](https://github.com/spring-projects/spring-security/milestones) out June 6th

**Comment by tha2015** [comment]
Thanks Rob.

**Comment by rwinch** [comment]
@jgrandja I'm reassigning this to you. To ensure the sample changes to Boot 2.0.0.M2 you will need to delete the samples/gradle folder. You can find some notes on the[ Boot Wiki about migration](https://github.com/spring-projects/spring-boot/wiki/Relaxed-Binding-2.0#migrating-from-1x-to-20).

**Comment by jgrandja** [comment]
@tha2015 The Boot samples have been updated to 2.0.0.M1

**Comment by tha2015** [comment]
Thanks Joe. I will try it tonight.

**Comment by tha2015** [comment]
Thanks, it works for my sample application which using Auth0 OpenID Connect(I copied several files from the sample to my app). There was one exception (IllegalArgumentException: nameAttributeKey cannot be empty) at OAuth2LoginAutoConfiguration.java:97 because it could not find the key 'user-info-name-attribute-key' for github. I resolved it by deleting all keys under  github, facebook, okta from oauth2-clients-defaults.yml and everything seems OK now.
